packages/components/src/Page/Page.tsx

```diff
@@ -59,10 +58,35 @@ export interface PageProps {
   readonly moreActionsMenu?: SectionProps[];
 }
 
+interface PagePropsWithIntro extends PageFoundationProps {
+  /**
+   * Content of the page. This supports basic markdown node types
+   * such as `_italic_`, `**bold**`, and `[link name](url)`.
+   */
+  readonly intro: string;
+
+  /**
+   * Causes any markdown links in the `intro` prop to open in a new
+   * tab, i.e. with `target="_blank"`.
+   *
+   * Can only be used if `intro` prop is also specified.
+   *
+   * Defaults to `false`.
+   */
+  readonly externalIntroLinks?: boolean;
+}
+
+interface PagePropsNoIntro extends PageFoundationProps {
+  readonly intro?: undefined;
+  readonly externalIntroLinks?: never;
+}
+export type PageProps = XOR<PagePropsWithIntro, PagePropsNoIntro>;
```

Comment:
This `PagePropsNoIntro` shouldn't be needed. We should be able to leave PageFoundationProps the way it was and just do something like:

```tsx
interface PageWithIntroProps extends PageFoundationProps {
  readonly intro: string;
  readonly externalIntroLinks?: boolean;
}

export type PageProps = XOR<PageFoundationProps, PageWithIntroProps>;
```

In this case we say that `PageFoundationProps` has an optional `intro` prop.

With the `XOR` we are saying that if the optional `externalIntroLinks` is used, the type will be `PageWithIntroProps` which will mean that `intro` is required.